service_capacity_modeling/interface.py

```diff
@@ -770,112 +770,76 @@ class BufferComponent(str, Enum):
     the Buffers interface itself (should be str).
     """
 
-    # "Traffic" related buffers, e.g. CPU and Network
+    # [Query Pattern] a.k.a. "Traffic" related buffers, e.g. CPU and Network
     compute = "compute"
-    # "Dataset" related buffers, e.g. Disk and Memory
+    # [Data Shape]    a.k.a. "Dataset" related buffers, e.g. Disk and Memory
     storage = "storage"
-    # Specific resources
+
+    # Resource specific component
     cpu = "cpu"
     network = "network"
     disk = "disk"
-    # Memory can be used for various buffers
     memory = "memory"
-    memory_traffic = "memory-traffic"
-    memory_write_cache = "memory-write-cache"
-    memory_read_cache = "memory-read-cache"
+
+
+class BufferIntent(str, Enum):
```

Comment:
You might want some comments describing what these mean.

Reply:
Yeah I think I'll document more when I implement the different intents, just wanted to get eyes on the API a bit (and I do document a bit in the derived section below).

Reply:
Added some comments.